controller.rb

```diff
@@ -1,130 +1,132 @@
-		@initbalance = Dialog::account_balance
-		@user = BankAccount.new(@user, @initbalance)
-		choice
-		elsif @answer == "no"
-			Dialog::goodbye
-		else
-			greeting
-		end
-	end
-	
-	def choice	
-		### Asks what you want you want to do now that your bank account exists.
-		@action = Dialog::how_can_we_help_you
-
-		### If the balance is chosen. Show current bank account balance with BankAccount method "show_balance"
-		if @action == "balance" 
-			@bpin = Dialog::enter_pin
-			@user.show_balance(@bpin)
-			choice
-
-		### If 'deposit' is chosen, prompt user for PIN number and deposit amount and deposit it in account using BankAccount method "deposit".
-		elsif @action == "deposit" 
-			@dpin = Dialog::enter_pin
-			@damount = Dialog::deposit_amount
-			@user.do_deposit(@dpin, @damount)
-			@user.show_balance(1234)
-			choice
-
-		### If 'withdraw' is chosen, prompt user for PIN number and withdraw amount and withdraw amount using BankAccount method "withdraw".
-		elsif @action == "withdraw" 
-			@wpin = Dialog::enter_pin
-			@wamount = Dialog::withdraw
-			@user.do_withdraw(@wpin, @wamount)
-			@user.show_balance(1234)
-			choice
-
-		### If 'leave' is chosen, say goodbye with BankAccount method "goodbye_cust".
-		elsif @action == "leave"
-			Dialog::goodbye_cust
-		else
-			Dialog::wrong_entry
-			choice
-		end
-	end
+  attr_accessor :bank
+
+  def initialize
+    @bank = Bank.new
+  end
+
+  def signup_signin # Asks a user if they have a pre-existing account.
+    response = Dialog::new_or_old_user
+
+    if response == "yes" 
+      signin
+
+    elsif response == "no"
+      make_an_account
+
+    else
+      signup_signin
+    end
+  end
+
+  def signin
+    @name = Dialog::signin_name
+    @pin = Dialog::signin_pin
+    if bank.name_exists?(@name, "customers") == true && bank.verify_pin(@name, @pin, "customers") == true
+      account_choice
+
+    else
+      Dialog::wrong_username_or_pin
+      signup_signin
+    end
+  end
+
+  def make_an_account # Asks user if they want to make an account.
+    answer = Dialog::greeting
+
+    if answer == "yes"
+      @name = Dialog::account_name
+      @pin = Dialog::enter_pin
+
+      if bank.name_exists?(@name, "customers") == false
+        @customer = Customer.new(bank.db, @name, @pin)
+        @customer.add_to_db
+        @customer_id = @bank.find_customer_id(@name, @pin) 
+        @account = Account.new(bank.db, @customer_id)
+        @account.save_to_db 
+        account_choice
+
+      else
+        Dialog::existing_account_error
+        signin
+      end
+
+    elsif answer == "no"
+      Dialog::goodbye
+
+    else
+      make_an_account
+    end
+  end
+
+  def account_choice # Asks user if they want to deposit/withdraw, view current balance, or end their session.
+    action = Dialog::how_can_we_help_you
+
+    if action == "balance"
+      Dialog::space
+      acct_list = @bank.customer_accounts(@name, @pin)
+      accounts = acct_list[0]["account_id"].to_s
```

Comment:
Selecting the 0 index of acct_list means you're always only selecting the first account in the list to be displayed. Currently, there is no way through the user interface to make add more than one account under the same customer, but the functionality is all there, the user just needs a way to do it. As a step in that direction, let's prepare this account listing part to at least display all of the accounts returned from the SQL query by iterating through each of them (even if it's only one) and displaying (for now) just the account_id of each.

Delete this line and then also make suggested changes in other comments below.